GoogleChromeLabs/ps-analysis-tool
# Feature: Use provider for tabs

## Description
This PR adds a context provider for the tabs component to enable easy data access across child components, data sharing, and data storage.


## Relevant Technical Choices


## Testing Instructions
- Open PSAT and navigate to the topics panel.
- Try navigating to sub panels and it should not cause any error.
- Now try with PA and RWS panels too. 


## Additional Information:



## Screenshot/Screencast



---

## Checklist






Fixes #
